### ISSUE TITLE
Arkouda spack package development

### ISSUE THREAD

**Issue by arezaii** [Bears-R-Us/arkouda#3891]
Since all the arkouda dependencies can be found in spack, the next logical task is to create a spack package for arkouda. This issue is to track the features and status of the arkouda spack package development in its initial stages. 

### Status
I have started development on two spack packages for arkouda, one for the server and one for the client. Having separate packages is the recommendation from the spack maintainers when you have a server and client and want them to be available independently. 

The server package is named `arkouda` and the python client is `py-arkouda`.

The `arkouda` package has one variant, `distributed` which controls whether we build a version suitable for multiple locales or just a single locale version. The default is to build for a single locale. 

The `py-arkouda` package has no variants. I have heard that the next release of the arkouda client will require the chapel python bindings, but these are not present in the current chapel spack package. I have started adding the python bindings as a variant to the chapel package to support the arkouda client package. 

### Testing
Testing so far is limited but I have installed the arkouda package on an IB system with slurm and successfully started a server on two nodes, then connected to it using the arkouda python client running on a third node. Similarly I have demonstrated using a single workstation for the server and client.

### Feedback
If there are other installation or build options others would like to see in the arkouda server or client packages, please note them here. I plan to make the first PR to spack in the coming weeks and would like to incorporate feedback by then. Thanks!

**Comment by arezaii** [comment]
The PR to add `arkouda` (server) and `py-arkouda` (client) is up as a draft at https://github.com/spack/spack/pull/48054. It relies on changes to the Chapel package that will go in with the Chapel 2.3 update at https://github.com/spack/spack/pull/48053. 
I expect both of these to be ready for review this week. Once they are accepted and merged the new Arkouda packages will be available immediately on Spack's `develop` branch, then they should be picked up for the next Spack release.

**Comment by arezaii** [comment]
The Spack packages have been made ready for review!

**Comment by arezaii** [comment]
The Spack packages have been merged into the develop branch of spack

**Comment by bradcray** [comment]
Nice!  Thanks for putting in the effort here, Ahmad!  Can this issue be closed, or is there more that you think should happen before doing so?
